google/cloud/aiplatform/training_jobs.py

```diff
@@ -2673,6 +2673,455 @@ def _model_upload_fail_string(self) -> str:
+                Whether to override the contents of [export_evaluated_data_items_bigquery_destination_uri],
+                if the table exists, for exported test set predictions. If False, and the
+                table exists, then the training job will fail.
+
+                Applies only if [export_evaluated_data_items] is True and
+                [export_evaluated_data_items_bigquery_destination_uri] is specified.
+            quantiles (List[float]):
+                Quantiles to use for the `minizmize-quantile-loss`
+                [AutoMLForecastingTrainingJob.optimization_objective]. This argument is required in
+                this case.
+
+                Accepts up to 5 quantiles in the form of a double from 0 to 1, exclusive.
+                Each quantile must be unique.
+            validation_options (str):
+                Validation options for the data validation component. The available options are:
+                "fail-pipeline" - (default), will validate against the validation and fail the pipeline
+                                  if it fails.
+                "ignore-validation" - ignore the results of the validation and continue the pipeline
+            budget_milli_node_hours (int):
+                Optional. The train budget of creating this Model, expressed in milli node
+                hours i.e. 1,000 value in this field means 1 node hour.
+                The training cost of the model will not exceed this budget. The final
+                cost will be attempted to be close to the budget, though may end up
+                being (even) noticeably smaller - at the backend's discretion. This
+                especially may happen when further model training ceases to provide
+                any improvements.
+                If the budget is set to a value known to be insufficient to train a
+                Model for the given training set, the training won't be attempted and
+                will error.
+                The minimum value is 1000 and the maximum is 72000.
+            model_display_name (str):
+                Optional. If the script produces a managed AI Platform Model. The display name of
+                the Model. The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+
+                If not provided upon creation, the job's display_name is used.
+            sync (bool):
+                Whether to execute this method synchronously. If False, this method
+                will be executed in concurrent Future and any downstream object will
+                be immediately returned and synced when the Future has completed.
+        Returns:
+            model: The trained AI Platform Model resource or None if training did not
+                produce an AI Platform Model.
+        """
+
+        training_task_definition = schema.training_job.definition.automl_forecasting
+
+        training_task_inputs_dict = {
```

Comment:
You should use the EPCL Python types now instead of a dictionary, unless Forecasting has some special restrictions.

```
from google.cloud.aiplatform.v1beta1.schema.trainingjob import (
    definition_v1beta1 as training_job_inputs,
)

...

training_job_inputs.AutoMlForecastingInputs(...)
```

Reply:
Hmm...how would that work? Seems the `_run_job` method expects a `dict`: https://github.com/googleapis/python-aiplatform/blob/dev/google/cloud/aiplatform/training_jobs.py#L265

Reply:
Yesterday's merge of https://github.com/googleapis/python-aiplatform/pull/220 changed it to `training_task_inputs: Union[dict, proto.Message]`.

However, I think Forecasting has had the EPCL removed so you might be restricted to using the dict for now.